File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/build_execution_deployment/build_tools/maven/MavenImportingSettings.java
package com.github.alexandrecarlton.idea.settings.layout.build_execution_deployment.build_tools.maven;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableMavenImportingSettings.class)
public interface MavenImportingSettings {
  Optional<Boolean> importMavenProjectsAutomatically();
  Optional<String> vmOptionsForImporter();
}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/build_execution_deployment/build_tools/maven/MavenImportingSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.build_execution_deployment.build_tools.maven

import com.fasterxml.jackson.annotation.JsonProperty

data class MavenImportingSettings(

    @JsonProperty("Import Maven projects automatically")
    val importMavenProjectsAutomatically: Boolean? = null,

    @JsonProperty("VM options for importer")
    val vmOptionsForImporter: String? = null
)
